salt/cli/daemons.py

```diff
@@ -436,10 +432,11 @@ def prepare(self):
 
             super(YourSubClass, self).prepare()
         """
-        super(ProxyMinion, self).prepare()
+        super().prepare()
 
-        if not self.values.proxyid:
-            self.error("salt-proxy requires --proxyid")
+        ## allow for native minion
```

Comment:
Why exactly was this commented out?

Reply:
@s0undt3ch It was interfering with the native minion, maybe I need to go back and change that to be surrounded by a is_junos, which was added long after the --proxyid check was removed.

The native minion leverages the proxy code with a direct connection into using ncclient to adjust the working of the Junos OS on the router, but it does not have a proxy id  since that is assocaited with using Pillar which the native minion does not use.

Created https://github.com/saltstack/salt/issues/59020 to fix this